app/components/add-cron-job.js

```diff
@@ -1,45 +1,61 @@
 import Component from '@ember/component';
 import { inject as service } from '@ember/service';
-import { task } from 'ember-concurrency';
+import { computed } from '@ember/object';
+import { task, timeout } from 'ember-concurrency';
+import { mapBy } from '@ember/object/computed';
+import Branch from 'travis/models/branch';
+import config from 'travis/config/environment';
 
 export default Component.extend({
   store: service(),
 
   classNames: ['form--cron'],
 
+  intervals: ['monthly', 'weekly', 'daily'],
+
+  options: ['Always run', 'Do not run if there has been a build in the last 24h'],
+
+  currentCronJobsBranches: mapBy('repository.cronJobs', 'branch.name'),
+
+  init() {
+    this.reset();
+    this._super(...arguments);
+  },
+
   reset() {
     this.setProperties({
       selectedBranch: null,
-      selectedInterval: null,
-      disable: null
+      selectedInterval: this.intervals.firstObject,
+      selectedOption: this.options[0]
     });
   },
 
-  save: task(function* () {
-    const store = this.get('store');
-    const repoId = this.get('branches.firstObject.repoId');
-    const branch = this.get('selectedBranch') || this.get('branches.firstObject');
+  comparedBranches: computed('currentCronJobsBranches', 'branches', function () {
+    if (!this.branches) { return []; }
 
-    const existingCrons = yield store.filter('cron', { repository_id: repoId }, (c) => {
-      c.get('branch.repoId') === repoId && c.get('branch.name') === branch.get('name');
+    let branchNames = this.get('currentCronJobsBranches');
+    this.branches.forEach((branch) => {
+      branch.set('disabled', branchNames.indexOf(branch.name) > -1);
     });
+    return this.branches;
+  }),
 
-    if (existingCrons.get('firstObject')) {
-      store.unloadRecord(existingCrons.get('firstObject'));
-    }
-
-    const cron = store.createRecord('cron', {
-      branch,
-      interval: this.get('selectedInterval') || 'monthly',
-      dont_run_if_recent_build_exists: this.get('selectedOption') || false
+  performSearchRequest: task(function* (query) {
+    yield timeout(config.intervals.searchDebounceRate);
+    this.set('loadingData', true);
+    this.setProperties({
+      branches: yield Branch.search(this.store, query, this.repository.id),
+      loadingData: false
```

Comment:
About `branch.set('disabled', branchNames.indexOf(branch.name) > -1);`
Like I said from UX perspective better solution is show disabled option than nothing.
About map to {name, disabled} object there is no sense because it will require additional method to find branch object by name

Reply:
It seems like your comment belongs to [this conversation](https://github.com/travis-ci/travis-web/pull/2024#discussion_r286672639). Let's keep the further conversation there.